src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -330,13 +334,7 @@ private set
 
                 lock (_lock)
```

Comment:
Why do we need these lock? I deleted it and all tests are green.

Reply:
I suggest to remove it at all. Also remove private setter for ` public IGraphType this[string typeName]` and inline its call to `  SetGraphType(typeName, value);` in `ConvertTypeReference`.

Reply:
And even remove
```
if (type == null)
                    throw new ArgumentOutOfRangeException(nameof(type), "A type is required to lookup.");
```
check since it's private method.

Reply:
And maybe change `  private IGraphType this[Type type] ` indexer to method like `Find(Type type)` because it's always hard to search indexer usages`.

Reply:
The locks should not be needed. Types cannot be added after SchemaTypes is initialized and a dictionary can be read from concurrently as long as there are no writers

Reply:
Seems to be a good consequence of refactoring.

Reply:
I would like to spend some extensive time cleaning up SchemaTypes. The suggested changes are all private and do not affect v4 release.

Reply:
I might rewrite the whole thing. There’s a lot of duplicate code in the constructor alone! But maybe it’s all fine and just needs a few tweaks

Reply:
If you have the code open already, go ahead and push some cleanup to it. I’ll be busy until this weekend.

Reply:
Done